addons/sale/models/sale_order.py

```diff
@@ -57,6 +57,7 @@ def _rec_names_search(self):
 
     company_id = fields.Many2one(
         comodel_name='res.company',
+        inverse='_inverse_company_id',
```

Comment:
This field is not computed, it makes no sense to specify an inverse :thinking:

Reply:
I suggested this change. The inverse is triggered with an `api.onchange`. It's based on the code on `account.move`.

If the use of a fallback is preferred, then doesn't it make more sense to specify `required=False` on `company_id`?

Reply:
```suggestion

```
perhaps removing the `inverse` parameter, and renaming  `_inverse_company_id` to `_onchange_company_id` would be better in that case?

Reply:
> I suggested this change. The inverse is triggered with an `api.onchange`. It's based on the code on `account.move`.

There is no reason to specify it as inverse on the field.  It's an onchange, triggered thanks to the `api.onchange` decorator.

> If the use of a fallback is preferred, then doesn't it make more sense to specify `required=False` on `company_id`?

Certainly not !
We want a company on the SO, but the user should not have a traceback if he empties the field temporarily.

Reply:
I don't see any reason to unset the company, but it seems that some PO is complaining about that :man_shrugging: .
I don't know if a specific error message for that situation is worth it, but it's probably the cleanest solution indeed (and clearer for the end user).

Reply:
So would it be sufficient to bring the diff down to just this ?
```python
    @api.onchange('company_id')
    def _onchange_company_id(self):
        for order in self:
            # This can't be caught by a python constraint as constraints are only triggered at save and 
            # there are compute methods that require this data to be set correctly before saving.
            if not order.company_id:
                raise ValidationError(_("We can't leave this document without any company. Please select a company for this document."))
```
and, in the .pot 
```
#. module: sale
#. odoo-python
#: code:addons/sale/models/sale_order.py:0
#, python-format
msgid ""
"We can't leave this document without any company. Please select a company "
"for this document."
msgstr ""
```

Reply:
Hello @Feyensv & @dbkosky ! Thank you for initiating a discussion. I made the changes accordingly. Can it be double-checked?